backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHListenerApi.scala

```diff
@@ -62,6 +63,21 @@ class CHListenerApi extends ListenerApi with Logging {
         s".local_engine.settings.log_processors_profiles",
       "true")
 
+    // add memory limit for external sort
```

Comment:
would you please explain the logic of this code?

Reply:
If in memory sort size is not set for SortingStep, then use this formula to assign one:
 (executor's OffHeap Memory Size / cores per executors) * 0.8